code/authenticator/RESTfulAPI_TokenAuthenticator.php

```diff
@@ -344,6 +319,40 @@ private function generateToken($expired = false)
     ); 
   }
 
+  /**
+   * Update the token of a token owner
+   * @param $owner          The token owner instance to update
+   * @param bool $expired   Set to true to generate an outdated token
+   * @return array|null     Token data array('token' => HASH, 'expire' => EXPIRY_DATE)
+   */
+  private function updateToken($owner, $expired = false)
+  {
+    $ownerId = null;
+    $tokenData = null;
+
+    // DB field names
+    $tokenDBColumn  = $this->tokenConfig['DBColumn'];
+    $expireDBColumn = $this->tokenConfig['expireDBColumn'];
+
+    do {
+      try {
+        //generate token
+        $tokenData = $this->generateToken( $expired );
+
+        // ensure we never regenerate the same token!
+        if($owner->{$tokenDBColumn} != $tokenData['token']){
+          $owner->{$tokenDBColumn}  = $tokenData['token'];
+          $owner->{$expireDBColumn} = $tokenData['expire'];
+          $ownerId = $owner->write();
+        }
+      } catch(Exception $e){
+        $ownerId = null;
```

Comment:
Thought I would raise the question, but no idea if the ORM actually returns the relevant exception info... Might be good to investigate...